### PR TITLE
♻️ server: support multiple kyc inquiry templates

### PR DESCRIPTION
**Pull request by mainqueg**
## Summary by CodeRabbit

* **New Features**
  * Multi-template, scope-aware KYC flows with create/resume inquiries, session/OTL issuance, and identity document attachment.
  * Persona branching support for two onboarding templates (panda and manteca).

* **Improvements**
  * Better account/template resolution across scopes, legacy-KYC detection, and improved error/tracing.
  * Persona-versioned requests and added development-chain identifiers.

* **Tests**
  * Expanded end-to-end and unit tests covering scopes, templates, legacy flows, and edge cases.

* **Chores**
  * Several ramp/onboarding providers replaced with not-implemented placeholders.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

---

<a href="https://app.devin.ai/review/exactly/exa/pull/637">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

### CODE REVIEW

File: server/api/kyc.ts [hunk 1]
```diff
@@ -1,98 +1,221 @@
...
 
 const debug = createDebug("exa:kyc");
 Object.assign(debug, { inspectOpts: { depth: undefined } });
 
-const GetKYCQuery = object({ templateId: optional(string()), countryCode: optional(literal("true")) });
+const GetKYCQuery = object({
+  templateId: optional(string()), // TODO remove this after deprecate templateId query parameter
+  countryCode: optional(literal("true")),
+  scope: optional(picklist(["basic"])),
+});
```
[the remark this answers is not in the document]
Comment (mainqueg) [reply server/api/kyc.ts]: @coderabbitai This field value is enforced by the type system in the function that receives this parameter, so using any other value will result in a ts error.

File: server/utils/ramps/manteca.ts [hunk 2]
```diff
@@ -851,57 +766,18 @@ async function request<TInput, TOutput, TIssue extends BaseIssue<unknown>>(
...
-
-  const additionalData = safeParse(MantecaOnboarding, {
-    tin: personaAccount.attributes.fields.tin?.value,
-    gender: personaAccount.attributes.fields.sex_1?.value,
-    isnotfacta: personaAccount.attributes.fields.isnotfacta?.value,
-    termsAccepted: personaAccount.attributes.fields.manteca_t_c?.value,
-  });
-  if (!additionalData.success) throw new Error(ErrorCodes.BAD_KYC_ADDITIONAL_DATA);
+export function validateIdentification(_inquiry: InferOutput<typeof Inquiry>) {
+  throw new Error("not implemented");
```
[the remark this answers is not in the document]
Comment (mainqueg) [reply server/utils/ramps/manteca.ts]: @cursor The onramp endpoints and related code aren’t currently in use, so we can safely allow these responses and behaviours. The proper implementation will be addressed in a future PR.

File: server/utils/ramps/manteca.ts [hunk 3]
```diff
@@ -311,89 +307,8 @@ export async function getProvider(
...
-    ),
-    acceptTermsAndConditions(account.replace("0x", "")),
-  ]);
-
-  for (const result of results) {
-    result.status === "rejected" && captureException(result.reason, { extra: { account } });
-  }
+export async function mantecaOnboarding(_account: string, _credentialId: string, _templateId: string) {
+  await Promise.reject(new Error("not implemented"));
 }
```
[the remark this answers is not in the document]
Comment (mainqueg) [reply server/utils/ramps/manteca.ts]: @coderabbitai The onramp endpoints and related code aren’t currently in use, so we can safely allow these responses and behaviours. The proper implementation will be addressed in a future PR.

File: server/utils/ramps/manteca.ts [hunk 4]
```diff
@@ -851,57 +766,18 @@ async function request<TInput, TOutput, TIssue extends BaseIssue<unknown>>(
...
-  const additionalData = safeParse(MantecaOnboarding, {
-    tin: personaAccount.attributes.fields.tin?.value,
-    gender: personaAccount.attributes.fields.sex_1?.value,
-    isnotfacta: personaAccount.attributes.fields.isnotfacta?.value,
-    termsAccepted: personaAccount.attributes.fields.manteca_t_c?.value,
-  });
-  if (!additionalData.success) throw new Error(ErrorCodes.BAD_KYC_ADDITIONAL_DATA);
+export function validateIdentification(_inquiry: InferOutput<typeof Inquiry>) {
+  throw new Error("not implemented");
 }
```
[the remark this answers is not in the document]
Comment (mainqueg) [reply server/utils/ramps/manteca.ts]: @coderabbitai The onramp endpoints and related code aren’t currently in use, so we can safely allow these responses and behaviours. The proper implementation will be addressed in a future PR.

File: server/utils/persona.ts [hunk 5]
```diff
@@ -120,109 +171,227 @@ async function request<TInput, TOutput, TIssue extends BaseIssue<unknown>>(
...
+      }
+
+      const countryCode = basicAccount.output.data[0]?.attributes["country-code"];
+      const userIdClasses = basicAccount.output.data[0]?.attributes.fields.documents.value.map(
+        (document) => document.value.id_class.value,
+      );
+      if (!userIdClasses?.length) throw new Error(scopeValidationErrors.ACCOUNT_NOT_CREATED);
+      if (!countryCode) throw new Error(scopeValidationErrors.ACCOUNT_NOT_CREATED);
+      const allowedIds = getAllowedMantecaIds(countryCode);
+      if (!allowedIds) throw new Error(scopeValidationErrors.NOT_SUPPORTED);
```
[the remark this answers is not in the document]
Comment (mainqueg) [reply server/utils/persona.ts]: every account with basic scope (previously validated) should have at least one document with an idClass, if it's not the account wasn't correctly initializated and should be an internal error

File: server/utils/persona.ts [hunk 6]
```diff
@@ -296,3 +443,47 @@ export function headerValidator() {
...
+type IdClass = (typeof IdentificationClasses)[number];
+type Country = (typeof MantecaCountryCode)[number];
+type Allowed = { allowedIds: readonly IdClass[] };
+const allowedMantecaCountries = new Map<Country, Allowed>([
+  ["AR", { allowedIds: ["id", "pp"] }],
+  ["BR", { allowedIds: ["id", "dl", "pp"] }],
+  ...(DevelopmentChainId.includes(chain.id as DevelopmentChain)
+    ? ([["US", { allowedIds: ["dl"] }]] as const)
+    : ([] as const)),
+] satisfies (readonly [Country, Allowed])[]);
```
[the remark this answers is not in the document]
Comment (mainqueg) [reply server/utils/persona.ts]: not required so far
